CHANGELOG.md

```diff
@@ -13,6 +13,8 @@
 - `[jest-validate]` Allow `maxWorkers` as part of the `jest.config.js` ([#8565](https://github.com/facebook/jest/pull/8565))
 - `[jest-runtime]` Allow passing configuration objects to transformers ([#7288](https://github.com/facebook/jest/pull/7288))
 - `[@jest/core, @jest/test-sequencer]` Support async sort in custom `testSequencer` ([#8642](https://github.com/facebook/jest/pull/8642))
+- `[jest-worker]` Return a promise from `end()`, resolving with the information whether workers exited gracefully ([#8206](https://github.com/facebook/jest/pull/8206))
```

Comment:
Mark as breaking

Reply:
Still don't see how returning extra information that you previously didn't get access to is breaking, but I guess it's just a tag in the changelog...

Reply:
Changing the return type of a function is a breaking change

Reply:
If it was previously `void` though?
I mean I guess you can do
```ts
const x: void = end();
```
and it's no longer assignable, but that's a pretty low bar to define what a breaking change is IMO.
But fine I'll mark it.
Did you think about the other open conversation (throw vs reject)?

Reply:
The reason I consider it breaking is that IMO it should be a type error if you don't handle promises - https://github.com/typescript-eslint/typescript-eslint/tree/master/packages/eslint-plugin/docs/rules is nice, but (again, IMO) it shouldn't be a linter thing.

>Did you think about the other open conversation (throw vs reject)?

Yeah, I still think it should only do one or the other. https://github.com/sindresorhus/execa/pull/276#issuecomment-499582055

Reply:
Interesting, never seen [that document](https://www.w3.org/2001/tag/doc/promises-guide#always-return-promises) before. It's pre async await though, not sure if "should never synchronously throw errors, since that would force duplicate error-handling logic on the consumer" is that much of an issue now.

Reply:
You can send a PR and see if they agree with you 😀 

https://github.com/w3ctag/promises-guide/blob/135dddc2eeeed7034420ac1050ee66b78775ba77/index.bs#L106

Reply:
Might actually do that. I suspect they'll insist that usage without async-await is still a big problem and that fire-and-forget usage is not, but I'm wondering if there's anything else to consider.
In our case, I do think that fire-and-forget is an unusually big problem, because that's how you used to call the API when it returned `void`, so I'm sure that many people will never bother to do anything with the new Promise, and for those it would be better to throw.

Reply:
node will print an error if it rejects without `catch` (and the errors says in the future it'll crash the process just like unhandled error), so I don't think it'll make a difference either way for consumers in the fire-and-forget case